homeassistant/components/sensor/bmw_connected_drive.py

```diff
@@ -74,7 +63,27 @@ def name(self) -> str:
     @property
     def icon(self):
         """Icon to use in the frontend, if any."""
-        return self._icon
+        from bimmer_connected.state import ChargingState
+        vehicle_state = self._vehicle.state
+        charging_state = vehicle_state.charging_status in \
+            [ChargingState.CHARGING]
+
+        if self._attribute == 'mileage':
```

Comment:
You could make a constant dict that maps the attributes to the icon and lookup that here, if attribute is not `charging_level_hv`.